dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Meta.java

```diff
@@ -63,17 +68,31 @@
     @EJB
     DataFileServiceBean datafileService;
 
+    @EJB
+    private VariableServiceBean dataVariableService;
+
+    @EJB
+    private PermissionServiceBean permissionService;
+
+    @Inject
+    private EmbargoAccessService embargoAccessService;
+
     @EJB
     DatasetDao datasetDao;
 
+
     @Deprecated
     @Path("variable/{varId}")
     @GET
     @Produces({"application/xml"})
-
     public String variable(@PathParam("varId") Long varId, @QueryParam("exclude") String exclude, @QueryParam("include") String include, @Context HttpHeaders header, @Context HttpServletResponse response) /*throws NotFoundException, ServiceUnavailableException, PermissionDeniedException, AuthorizationRequiredException*/ {
         String retValue = "";
 
+        Optional<Dataset> datasetOptional = getDatasetFromDataVariable(varId);
+        if(datasetOptional.isPresent() && embargoAccessService.isRestrictedByEmbargo(datasetOptional.get())) {
```

Comment:
Optional is designed more for a functional style -> 

```
datasetOptional.
filter()
.ifPresent()

Reply:
To be honest this is less readable to me and I would rather drop optional usage than replace simple if statement with filtering out unwanted results and then checking if we are left with something to operate onto.
But i guess I can make it functional style like for exercise purposes.

Reply:
Hmm, well can't really argue if something is less readable for you :P. But in terms of dropping optional, it's even less readable since your function wouldn't say if value is always present or sometimes. So you would give less information.